### PR TITLE
feat: decouple tfy-agent and tfy-agent-proxy roles

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: charts/tfy-agent/templates/tfy-agent-clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.tfyAgent.enabled -}}
+{{- if .Values.tfyAgent.clusterRole.enable -}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "tfy-agent.fullname" . }}
```
Comment (akashg3627) [inline charts/tfy-agent/templates/tfy-agent-clusterrole.yaml:6]:
```suggestion
  name: {{ include "tfy-agent.fullname" . }}-cluster-role
```

File: charts/tfy-agent/templates/tfy-agent-clusterrolebinding.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.tfyAgent.enabled -}}
+{{- if .Values.tfyAgent.clusterRole.enable -}}
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ include "tfy-agent.fullname" . }}
```
Comment (akashg3627) [inline charts/tfy-agent/templates/tfy-agent-clusterrolebinding.yaml:6]:
```suggestion
  name: {{ include "tfy-agent.fullname" . }}-cluster-rolebinding
```

File: charts/tfy-agent/templates/tfy-agent-clusterrole.yaml (hunk 1 above)
Comment (debajyoti-truefoundry) [reply charts/tfy-agent/templates/tfy-agent-clusterrole.yaml:6]: Not required.

File: charts/tfy-agent/templates/tfy-agent-clusterrolebinding.yaml (hunk 2 above)
Comment (debajyoti-truefoundry) [reply charts/tfy-agent/templates/tfy-agent-clusterrolebinding.yaml:6]: Not required.

**Review by akashg3627 — APPROVED** [review]

**Pull request merged**